mmdet/models/utils/norm.py

```diff
@@ -1,17 +1,38 @@
 import torch.nn as nn
 
-norm_cfg = {'BN': nn.BatchNorm2d, 'SyncBN': None, 'GN': None}
+
+norm_cfg = {'BN': nn.BatchNorm2d, 'SyncBN': None, 'GN': nn.GroupNorm}
 
 
 def build_norm_layer(cfg, num_features):
+    """
+    cfg should contain:
+        type (str): identify norm layer type.
+        layer args: args needed to instantiate a norm layer.
+        frozen (bool): [optional] whether stop gradient updates
+            of norm layer, it is helpful to set frozen mode
+            in backbone's norms.
+    """
     assert isinstance(cfg, dict) and 'type' in cfg
     cfg_ = cfg.copy()
-    cfg_.setdefault('eps', 1e-5)
     layer_type = cfg_.pop('type')
+    frozen = cfg_.pop('frozen') if 'frozen' in cfg_ else False
+
+    # args name matching
+    if layer_type == 'GN':
+        assert 'num_groups' in cfg
+        cfg_.setdefault('num_channels', num_features)
+    elif layer_type == 'BN':
+        cfg_.setdefault('num_features', num_features)
+    cfg_.setdefault('eps', 1e-5)
 
     if layer_type not in norm_cfg:
         raise KeyError('Unrecognized norm type {}'.format(layer_type))
     elif norm_cfg[layer_type] is None:
         raise NotImplementedError
 
-    return norm_cfg[layer_type](num_features, **cfg_)
+    norm = norm_cfg[layer_type](**cfg_)
+    if frozen:
+        for param in norm.parameters():
+            param.requires_grad = False
+    return norm
```

Comment:
A suggestion to include both "frozen" and "eval_mode" in arguments and return the layer and abbreviation.
```python
norm_cfg = {
    'BN': (nn.BatchNorm2d, 'bn'),
    'SyncBN': (None, 'bn'),
    'GN': (nn.GroupNorm, 'gn'),
    # and potentially 'SN'
}


def build_norm_layer(cfg, num_features):
    assert isinstance(cfg, dict) and 'type' in cfg
    cfg_ = cfg.copy()
    layer_type = cfg_.pop('type')
    frozen = cfg_.pop('frozen', False)
    # eval_mode only has effects on BN and its variants
    eval_mode = cfg_.pop('eval_mode', False)

    if layer_type not in norm_cfg:
        raise KeyError('Unrecognized norm type {}'.format(layer_type))
    else:
        norm_layer, abbr = norm_cfg[layer_type]
        if norm_layer is None:
            raise NotImplementedError

    cfg_.setdefault('eps', 1e-5)
    if layer_type != 'GN':
        layer = norm_layer(num_features, **cfg_)
    else:
        assert 'num_groups' in cfg_
        layer = norm_layer(num_channels=num_features, **cfg_)

    if frozen:
        for param in layer.parameters():
            param.requires_grad = False

    if eval_mode:
        param.eval()

    return layer, abbr
```

Reply:
If set eval here, it will be overwritten on train() method of backbone.

Reply:
Oh I ignored that, so we still need to set eval mode in `resnet.train()`.

Reply:
abbr is a nice idea for backbones but might be not helpful at neck and roi_head norms. I think keeping abbr processed in backbones in ok, and build_norm_layer is more straightforward.

Reply:
In neck and heads, we usually use ConvModule instead of directly using norm layers. A ConvModule can be more clear if we also adopt "bn", "gn" instead of "norm".

Reply:
I totally agree. It will be fixed that after the sanity benchmark check.

Reply:
If the norm name is changed, says from 'norm' to 'bn',  existing pretrained models won't be loaded directly.

Reply:
Existing models in the model zoo have no norm layers in necks and heads, except for this GN model. We can rename keys of the trained GN model without much efforts.

Reply:
I see. It is updated in [3fdd041](https://github.com/open-mmlab/mmdetection/pull/160/commits/3fdd041cd7f4bd6fc877bac22b28448ae738033d)